hw4/01-routing.js

```diff
@@ -12,37 +14,60 @@ const port = process.env.PORT || 5001;
 
 // For other routes, such as http://localhost:5001/other, this exercise should return a status code 404 with '404 - page not found' in html format
 
-const routes = [
-  'welcome',
-  'redirect',
-  'redirected',
-  'cache',
-  'cookie',
-  'other',
-];
+const routes = ["welcome", "redirect", "redirected", "cache", "cookie", "other"];
 
 let getRoutes = () => {
-  let result = '';
+  let result = "";
 
-  routes.forEach(
-    (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
-  );
+  routes.forEach((elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`));
 
   return result;
 };
 
-app.get('/', (req, res) => {
+app.get("/", (req, res) => {
   let routeResults = getRoutes();
 
-  res.writeHead(200, { 'Content-Type': 'text/html' });
+  res.writeHead(200, { "Content-Type": "text/html" });
   res.write(`<h1>Exercise 04</h1>`);
   res.write(`<ul> ${routeResults} </ul>`);
   res.end();
 });
 
-app.get('/welcome', (req, res) => {});
+app.get("/welcome", (req, res) => {
+  res.status(200);
+  res.set({ "Content-Type": "text/html" });
+  res.send("Welcome to the Routing Exercise!!");
+});
+
+app.get("/redirect", (req, res) => {
+  res.redirect(302, "/redirected");
+});
+
+app.get("/redirected", (req, res) => {
+  res.status(200);
+  res.set({ "Content-Type": "text/plain" });
+  res.send("Your request to /redirect has been redirected to this page.");
+});
 
-// Add your code here
+app.get("/cache", (req, res) => {
+  res.status(200);
+  res.set("Cache-control", "public, max-age=86400");
+  res.set({ "Content-Type": "text/plain" });
+  res.send("This resource was cached");
```

Comment:
It should be in HTML format not in plain text as per HW4

Reply:
Thanks. I will make the changes accordingly.